src/main/scala/wordcounter/AkkaWordCounter.scala

```diff
@@ -101,51 +116,33 @@ class Listener extends Actor {
 
 object AkkaWordCounter extends App {
 
+  import RoutingActor.StartProcessFileMsg
+  import akka.pattern.ask
   import akka.util.Timeout
+
   import scala.concurrent.duration._
-  import akka.pattern.ask
-  import akka.dispatch.ExecutionContexts._
 
   override def main(args: Array[String]) {
-
-    val files = getListOfFiles("src/main/resources/")
-
-    /**
-      * foreach takes a procedure -- a function with a result type Unit -- as the right operand.
-      * It simply applies the procedure to each List element.
-      * The result of the operation is again Unit; no list of results is assembled.
-      */
-    files.foreach(initActorSystem)
-
+    val sourceDirectoryName = if (args.length > 0) args(1) else "src/main/resources/"
+    val directory = new File(sourceDirectoryName)
+    if (directory.exists && directory.isDirectory) {
+      val system = ActorSystem()
+      implicit val ec = system.dispatcher
+      val results = directory.listFiles.map(_.getAbsolutePath).map(processFile(system, _))
```

Comment:
I also moved away from waiting for the Future to return to terminate the system. I used a separate `Actor` to handle it.